### ISSUE TITLE
`Declaration isn't referenced` review for private members

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4822]
**Describe the issue**
In the below example, I'd like for someone to please explain to me why this lint is being triggered in `N._fn`, since it is a protected method that can only be called from within a library, this lint should potentially know that.

Also, in a class like `C`, I'm not sure this should be triggered. However, I'm not aware if this lint can be aware of other members in the super method it is being overridden, because if it is aware, of them, then in `C` I don't believe it should be triggered. Still, probably in `D` but that would also bring a question as to why would this be called and not maybe warn that the mixin is unused instead.

Maybe the `Unused-Mixin` could be called when the mixin is a private mixin like in `E`?

**To Reproduce**
```dart
mixin _M {
  void _fn();
}

mixin M {
  void _fn(); // The declaration '_fn' isn't referenced.
}

mixin N implements M {
  @override
  void _fn(); // The declaration '_fn' isn't referenced.
  void a() {}
}

class A with N {
  void fn() {
    _fn();
  }
  
  @override
  void _fn() {}
}

class B with N { } // Missing concrete implementation of 'N._fn'.

class C with N {
  void _fn() {} // The declaration '_fn' isn't referenced.
}

class D with M {
  @override
  void _fn() {} // The declaration '_fn' isn't referenced.
}

class E with _M {
  @override
  void _fn() {}
}
```

**Additional context**
I'm really out of my depth here, but maybe someone can help me make my questions clearer.

**Comment by srawlins** [comment]
I'm a little confused about the question, but we can dig in. `N._fn` is never called. Can you point to a spot where it is called?

`C._fn` is also never called. Can you expand your repro to a file where `C._fn` is called, but the warning is still reported?

I don't really understand what your comment is about `D`. It looks identical to `C` (but with an annotation, which does not come into play).

The thread gets too spread out for me after that. 😄 . I won't address your comments about `E` here (or, yet), to keep the issue to one topic.
